src/hoc/helpers.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from "react";
+
+const getDisplayName = (prefix: string, component: React.ComponentType) => {
+  return `${prefix}(${component.displayName || component.name || "HOC"})`;
+};
+
+/**
+ * Helper function for creating HOC which will inject context as a prop.
+ * @param displayName - name for easier debugging
+ * @param injectedPropName - under what prop context should be injected
+ * @param Context - context returned by React.createContext()
+ */
+const createHOCFromContext = <Props extends {}>(
```

Comment:
Please do not introduce new HOC. We use the render-as-a-function pattern (sometimes called "render prop" but we always use `children`) and plan to move to hooks when they become available in stable builds.